test/e2e/dsa/dsa.go

```diff
@@ -96,4 +98,28 @@ func describe() {
 
 		framework.Logf("log output: %s", log)
 	})
+
+	ginkgo.It("deploys DSA plugin with operator", func() {
+		utils.Kubectl(ns, "apply", "-k", "deployments/operator/e2e/kustomization.yaml")
+
+		if _, err := e2epod.WaitForPodsWithLabelRunningReady(f.ClientSet, ns, labels.Set{"control-plane": "controller-manager"}.AsSelector(), 1, timeout); err != nil {
+			framework.Failf("unable to wait for all pods to be running and ready: %v", err)
+		}
+
+		time.Sleep(timeout)
+
+		utils.Kubectl(ns, "apply", "-f", "deployments/operator/samples/deviceplugin_v1_dsadeviceplugin.yaml")
+
+		if _, err := e2epod.WaitForPodsWithLabelRunningReady(f.ClientSet, ns, labels.Set{"app": "intel-dsa-plugin"}.AsSelector(), 1, timeout); err != nil {
+			framework.Failf("unable to wait for all pods to be running and ready: %v", err)
+		}
+
+		if err := utils.WaitForNodesWithResource(f.ClientSet, "dsa.intel.com/wq-user-dedicated", timeout); err != nil {
+			framework.Failf("unable to wait for nodes to have positive allocatable resource: %v", err)
+		}
+
+		utils.Kubectl(ns, "delete", "-f", "deployments/operator/samples/deviceplugin_v1_dsadeviceplugin.yaml")
+
+		f.DeleteNamespace(ns)
```

Comment:
>Maybe that's causing issues?
Theoretically yes, however not in this case, i.e. we have a clean run for each worklflow.

Everything succeeds, so namespace isn't a problem here too ("ns" is propagated to RunKubectlOrDie()).

An unreliable webhook availability in a CPU constrained environent is a problem.

Ther previous problems were caused by SGX's AfterEach which is fixed.

Reply:
You could try if something like this would help instead:
```
-	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+	if err := mgr.AddReadyzCheck("readyz", mgr.GetWebhookServer().StartedChecker()); err != nil {
```

Reply:
For the record, `utils.TestWebhookServerTLS()` is not going to be acceptable.